src/plugins/intel_npu/src/plugin/npuw/partitioning/online/snapshot.cpp

```diff
@@ -723,6 +715,20 @@ std::shared_ptr<Repeated> Snapshot::tryMergeTriangles(const std::vector<Group::G
         return {};
     }
 
+    if (prods.size() < m_ctx.keep_blocks) {
+        // In some cases (specifically mixed precision) during MergeUniques() pass we could be left with
+        // E.g. 10 repeated blocks with tag AAA and 2 repeated blocks with tag BBB
```

Comment:
The missing part for this great comment is to indicate that the structure may look like

```
AAA -> AAA -> AAA -> BBB -> AAA -> AAA -> AAA -> BBB
```

So our algorithm won't merge `AAA + AAA` but will keep merging `AAA + (AAA)BBB` so the RHS will grow enormously